actors/badge/src/main/java/org/sunbird/badge/actors/BadgeAssociationActor.java

```diff
@@ -0,0 +1,206 @@
+    List<String> associationIds = getAssociationIdsToBeRemoved(activeBadges, reqestedBadges);
+    List<Map<String, Object>> updateMapList = new ArrayList<>();
+    Response response = new Response();
+    if (CollectionUtils.isNotEmpty(associationIds)) {
+      // boolean flag = ContentService.updateEkstepContent(contentId,
+      // BadgingJsonKey.BADGE_ASSOCIATIONS, activeBadges);
+      updateMapList = updateCassandraAndGetUpdateMapList(associationIds, requestedBy);
+    }
+    sender().tell(response, self());
+    associationService.syncToES(updateMapList, false);
+  }
+
+  private List<Map<String, Object>> updateCassandraAndGetUpdateMapList(
+      List<String> associationIds, String requestedBy) {
+    List<Map<String, Object>> updateList = new ArrayList<>();
+    for (String id : associationIds) {
+      Map<String, Object> updateMap =
+          associationService.getCassandraBadgeAssociationUpdateMap(id, requestedBy);
+      updateList.add(updateMap);
+      contentBadgeAssociationDao.updateBadgeAssociation(updateMap);
+    }
+    return updateList;
+  }
+
+  private List<String> getAssociationIdsToBeRemoved(
+      List<Map<String, Object>> activeBadges, List<String> reqestedBadges) {
+    // TODO Auto-generated method stub
+    List<String> badgeIds = getNewBadgeIdsToBeAdded(reqestedBadges, activeBadges);
+    if (CollectionUtils.isNotEmpty(badgeIds)) {
+      ProjectCommonException.throwClientErrorException(ResponseCode.errorBadgeAssociationNotFound);
+    }
+    List<String> associationIds = new ArrayList<>();
+    if (CollectionUtils.isNotEmpty(activeBadges)) {
+      for (int i = 0; i < activeBadges.size(); i++) {
+        if (reqestedBadges.contains((String) activeBadges.get(i).get(BadgingJsonKey.BADGE_ID))) {
+          String associationId = (String) activeBadges.get(i).get(BadgingJsonKey.ASSOCIATION_ID);
+          associationIds.add(associationId);
+          activeBadges.remove(i);
+        }
+      }
+    }
+    return associationIds;
+  }
+
+  private Map<String, Object> getContentDetails(Request request) {
+    Map<String, String> headers = CourseBatchSchedulerUtil.headerMap;
+    Map<String, Object> contentDetails =
+        CourseEnrollmentActor.getCourseObjectFromEkStep(
+            (String) request.getRequest().get(JsonKey.CONTENT_ID), headers);
+    if (MapUtils.isEmpty(contentDetails)) {
+      ProjectCommonException.throwClientErrorException(ResponseCode.invalidContentId);
+    }
+    return contentDetails;
+  }
+
+  private List<Map<String, Object>> getBadgesDetailsToBeAdded(
+      List<Map<String, Object>> activeBadgesList, List<String> requestedBadges) {
+    List<String> newBadgeIdsList = getNewBadgeIdsToBeAdded(requestedBadges, activeBadgesList);
+    List<Map<String, Object>> newBadgesDetails = getBadgesDetails(newBadgeIdsList);
+    return newBadgesDetails;
+  }
+
+  private List<String> getNewBadgeIdsToBeAdded(
```

Comment:
getBadgeIdNotIn - as this is used in two context - once while adding other while finding - if remove list has some badge- which does not exist